### PR TITLE
Display Quick Action Button

### PR DESCRIPTION
**Pull request by Gonals**
### Fixed Issues

$ https://github.com/Expensify/App/issues/38050
PROPOSAL:

### Tests
Let's do this!

1. Log into newDot from a new account
2. Open the Global Create menu. Confirm no additional item is displayed:
Screenshot 2024-03-20 at 3 28 53 PM

4. Request Money (manual) from someone
5. After completing the request, confirm it displays as a quick action in the Global create menu:
Screenshot 2024-03-20 at 3 29 46 PM

6. Confirm You can complete the request from the quickAction, and that it points to the same user the original request did.
7. Repeat the same steps, but requesting with a receipt:
Screenshot 2024-03-20 at 3 32 46 PM

8. And again, but with distance:
Screenshot 2024-03-20 at 3 33 29 PM

9. Confirm all of them work for Money requests from workspaces too:
Screenshot 2024-03-20 at 11 27 55 AM

10. Confirm all of them work for Split requests too.
11. Send money to someone and, again, confirm it later displays as a a quick action and that it works:
Screenshot 2024-03-20 at 3 36 26 PM

11. Assign a task to someone and confirm it later displays as a a quick action and that it works:
Screenshot 2024-03-20 at 3 40 48 PM

- [x] Verify that no errors appear in the JS console

### Offline tests
Same, but offline

### QA Steps
Same as tests

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Screenshot 2024-03-20 at 5 19 53 PM

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>
Screenshot 2024-03-20 at 5 41 48 PM

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-03-20 at 4 44 30 PM

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-03-20 at 4 57 08 PM

</details>

### CODE REVIEW

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/languages/es.ts
Comment (Gonals) [reply src/languages/es.ts]: True. Fixed

File: src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js
Comment (Gonals) [reply src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js:380]: What do you mean?

**Comment by trjExpensify** [comment]
@shawnborton @dannymcclain @dubielzyk-expensify anyoneee wanna' take the quick action button for a spin?

**Comment by shawnborton** [comment]
Nice, will build it now!

**Comment by shawnborton** [comment]
I also don't think "Shortcut" should appear as part of the row :hover - it is basically a label that sits on top of the row, so I wouldn't think it would be included when hovering over the Request money row.

**Comment by shawnborton** [comment]
Looks like an incorrect border color when hovering over a split:
CleanShot 2024-03-20 at 13 32 54@2x

Also, I wasn't able to screenshot fast enough but there was a weird update happening when I had global create menu - I think the Request Money icon changed from a $ to arrows maybe?

**Comment by shawnborton** [comment]
Self DM is missing an avatar on the right side:
CleanShot 2024-03-20 at 13 34 32@2x

**Comment by shawnborton** [comment]
What size avatar are we choosing to show here? Shouldn't it be the 40x40 size? Right now it's smaller and it doesn't go to the right edge of the row:
CleanShot 2024-03-20 at 13 35 48@2x

**Comment by trjExpensify** [comment]
> What size avatar are we choosing to show here? Shouldn't it be the 40x40 size? Right now it's smaller and it doesn't go to the right edge of the row: CleanShot 2024-03-20 at 13 35 48@2x

Ah, I _think_ it might have been @dannymcclain that knocked it down to 24x24 in the predesigns. 🤔

**Comment by shawnborton** [comment]
Ah okay, but then the double/split avatars for groups is at 40x40?

What if instead we just use the same size avatars we use for compact LHN rows? It might be 28x28? I think we can pass that into the avatar component? And then this way we always have a consistent bounding box for the avatar at least.

**Comment by trjExpensify** [comment]
Works for me! I think some people were concerned initially that 40x40 was too big, but I actually think it's fine personally, so I wouldn't be against just sticking with that.

**Comment by shawnborton** [comment]
Oh I’m saying if we go with the #focus mode avatars, they will be smaller -
like 28x28 I think.

**Comment by dannymcclain** [comment]
>I think we should truncate long names to be just one line

>I also don't think "Shortcut" should appear as part of the row :hover - it is basically a label that sits on top of the row, so I wouldn't think it would be included when hovering over the Request money row.

Agree!

Also, I'm down for the focus mode size avatars, but I do maintain that 40x40 is too big. I also agree that we need to ditch that margin-right that's currently on the avatar.

**Comment by Gonals** [comment]
> I also don't think "Shortcut" should appear as part of the row :hover - it is basically a label that sits on top of the row, so I wouldn't think it would be included when hovering over the Request money row.

This is currently

> Oh I’m saying if we go with the #focus mode avatars, they will be smaller - like 28x28 I think.

This is how it looks in the next size (28x28):
Screenshot 2024-03-20 at 7 53 33 PM

**Comment by shawnborton** [comment]
Can you show that for more cases, like the group avatar version?

**Comment by Gonals** [comment]
> Can you show that for more cases, like the group avatar version?

Sure:
Screenshot 2024-03-20 at 8 10 52 PM
Screenshot 2024-03-20 at 8 10 29 PM

**Comment by Gonals** [comment]
> Also, I'm down for the focus mode size avatars, but I do maintain that 40x40 is too big. I also agree that we need to ditch that margin-right that's currently on the avatar.

Yep, 40 looks huge.

The margin right is the current style for MenuItems with avatars on the right, but I can likely remove it, particularly since I can't find anywhere else using them 🤷

**Comment by dannymcclain** [comment]
@Gonals in your screenshot for the split, the avatars look really off center—any idea why? I would expect them to just be vertically centered within the icon + title/participants row.

CleanShot 2024-03-20 at 14 26 02@2x

**Comment by dubielzyk-expensify** [comment]
Tested on iOS and nothing special stood out besides the issues @shawnborton highlighted

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/components/MenuItem.tsx
Comment (Gonals) [reply src/components/MenuItem.tsx]: Note that this is the only flow that actually uses floatRightAvatars, as far as I can tell, so we are pretty safe making changes.
This is getting removed, anyways, as I mentioned above

File: src/libs/actions/Task.ts
Comment (Gonals) [reply src/libs/actions/Task.ts:714]: AccountID defaults to 0

File: src/components/MultipleAvatars.tsx
Comment (Gonals) [reply src/components/MultipleAvatars.tsx:105]: This is an existing bug, as we should account for hovering by default.

**Comment by Gonals** [comment]
Comments addressed! I'm not sure what's up with the performance tests, though

**Comment by shawnborton** [comment]
Gonna spin up a new test build for us

**Comment by dannymcclain** [comment]
Hmm, I'm not seeing anything different in the new test build?

CleanShot 2024-03-20 at 13 55 12@2x
CleanShot 2024-03-21 at 09 17 00@2x

**Comment by Gonals** [comment]
> Hmm, I'm not seeing anything different in the new test build?
> 
> CleanShot 2024-03-20 at 13 55 12@2x CleanShot 2024-03-21 at 09 17 00@2x

Hmmm. Lemme check. Maybe something broke when merging main

**Comment by Gonals** [comment]
Nope. Seems fine to me:
Screenshot 2024-03-21 at 3 24 55 PM

**Comment by dannymcclain** [comment]
Ok I did a hard refresh on the ad hoc build and am seeing the updated hover style (looks good! we just need to make sure we change the border color on that avatar on hover). But whenever I try to create a money request the app crashes. Also it looks like the double-avatars are still not vertically centered.

CleanShot 2024-03-21 at 09 48 56@2x

**Comment by Gonals** [comment]
Ok, border fixed! @dannymcclain, I think I'm going to need detailed instructions of what you want with the avatars 😅. They look in place to me, but I am pretty terrible at `seeing` things

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Could you please add a new story for this change on storyBook?

**Comment by Gonals** [comment]
> Ok, border fixed! @dannymcclain, I think I'm going to need detailed instructions of what you want with the avatars 😅. They look in place to me, but I am pretty terrible at `seeing` things

For menuItem? It doesn't really have label/right avatars in our storybook. We can add them as a follow up, but I don't want to slow this PR on it

**Comment by Gonals** [comment]
> Could you please add a new story for this change on storyBook?

For menuItem? It doesn't really have label/right avatars in our storybook. We can add them as a follow up, but I don't want to slow this PR on it

**Comment by dannymcclain** [comment]
Ok hopefully this is helpful 😅  cc @Expensify/design for a gut check on these as well.

image

**Comment by Gonals** [comment]
> Ok hopefully this is helpful 😅 cc @Expensify/design for a gut check on these as well.
> 
> image

Very helpful. Thanks! Now to figure out how to get it done 😁

**Comment by Gonals** [comment]
Ok! What about this!
Screenshot 2024-03-22 at 12 18 20 AM
Screenshot 2024-03-22 at 12 18 42 AM

**Comment by dubielzyk-expensify** [comment]
The single avatar looks centered to me, but the split one doesn't look quite centered:
CleanShot 2024-03-22 at 09 40 26@2x

The arrow on the left also isn't centered properly

**Comment by Gonals** [comment]
> The single avatar looks centered to me, but the split one doesn't look quite centered: CleanShot 2024-03-22 at 09 40 26@2x
> 
> The arrow on the left also isn't centered properly

I don't think I want to touch the arrow on the left. All icons in menuItem use that margin styling, so I'd rather not mess with it 😅.

For the right, I `can` change it, but our standard margins are set in 4px intervals, so moving the bottom margin 2px seems kinda weird, but it is doable if we really want to.

Note that The top of split also has a different distance from the top that the bottom of the description has from the bottom, so this may be centered with the text?  Thoughts?

**Comment by dubielzyk-expensify** [comment]
Keen to hear what @Expensify/design thinks here. Feels like a inconvenient fix to something that has a bigger problem which is that the things aren't centered in the row as it should. To me, that row is more a flexbox container with the avatar being centered rather than padding from the top. But let's wait to hear what the designers think

**Comment by Gonals** [comment]
> To me, that row is more a flexbox container with the avatar being centered rather than padding from the top. But let's wait to hear what the designers think

And that does work fine with the single avatar, but the double one does not get centered correctly 🤷

**Comment by Gonals** [comment]
We can always stack them horizontally. That'll likely simplify this:
Screenshot 2024-03-22 at 1 07 28 AM

**Comment by Gonals** [comment]
Ok, I played around with it for a bit and got this. How's it looking?
Screenshot 2024-03-22 at 12 18 20 AM
Screenshot 2024-03-22 at 12 18 42 AM

**Comment by Gonals** [comment]
Ok, played around for a bit and got this. I think it is centered now 😅 :
Screenshot 2024-03-22 at 1 22 47 AM
Screenshot 2024-03-22 at 1 23 30 AM

**Comment by dubielzyk-expensify** [comment]
Eyyy, nice job. That last screenshot looks good to me 😄

**Comment by Gonals** [comment]
Great! Seems like I got the design folks happy (or close to that 🤣). All yours @parasharrajat!

**Comment by Gonals** [comment]
Conflicts fixed

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js
Comment (rlinoz) [inline src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js:111]: I don't see where we are using this, am I missing something?

**Review by rlinoz — COMMENTED** [review]
This feature is really cool and it is working really well, great work!

Just one question

File: src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js
Comment (trjExpensify) [reply src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js:111]: We'll be using that to determine if we show an educational tooltip about the quick action: https://github.com/Expensify/App/issues/38054

Comment (Gonals) [reply src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js:111]: Yep. Not in use yet, but it'll come with the follow-up PRs

File: src/styles/utils/spacing.ts
Comment (Gonals) [reply src/styles/utils/spacing.ts]: Oh, you are right! Fixed

**Comment by Gonals** [comment]
Comments addressed @parasharrajat, @rlinoz

**Review by rlinoz — DISMISSED** [review]

**Comment by Gonals** [comment]
> @Gonals I noticed that when we use split bill quick action, the user is navigated to the split bill chat instead, the user lands on LHN.

I'm not sure what you mean. When I use the split bill quick action, it takes me to the expected chat report

**Comment by Gonals** [comment]
> Somehow, chats are not loading for me currently and creating issues while testing.

Seems to work fine for me. Cn you give it another try? Let's see if we can get this merged today

**Review by parasharrajat — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.


:ribbon: :eyes: :ribbon:  C+ reviewed

**Comment by shawnborton** [comment]
Spinning up another adhoc build for testing before we merge this.

**Comment by Gonals** [comment]
> Spinning up another adhoc build for testing before we merge this.

Neat! Once this merges, It'll unblock the two remaining issues 😁

**Comment by shawnborton** [comment]
So for the group avatar, I see it in the menu like this:
CleanShot 2024-03-25 at 10 35 21@2x


But in the LHN, it looks like this:
CleanShot 2024-03-25 at 10 34 55@2x

So sorry if this wasn't clear, but the idea was to just reuse the exact avatar styles from the LHN in #focus mode. This way you don't need to write or refactor the avatar component, we already have an existing avatar component that can show a 28x28 frame with avatar(s) in it. 

Basically we can just reuse the exact styles here:
CleanShot 2024-03-25 at 10 36 24@2x


Can you update to match those styles please? Thanks!

**Comment by Gonals** [comment]
> So sorry if this wasn't clear, but the idea was to just reuse the exact avatar styles from the LHN in #focus mode. This way you don't need to write or refactor the avatar component, we already have an existing avatar component that can show a 28x28 frame with avatar(s) in it.

Gotcha. Taking a look on how to reuse that styling in this component.
Are we sure about that, though? I think the 28x28 looks a bit small for this 🤷

**Comment by shawnborton** [comment]
That would only be for the group avatars, and it matches what we have in the LHN, so I think we should do it. I just think it's going to simplify things so much that we don't have to maintain more avatar styles - basically we're choosing to reuse exactly what we do for the avatars in the LHN for #focus mode, and those styles already exist.

**Comment by Gonals** [comment]
> That would only be for the group avatars, and it matches what we have in the LHN, so I think we should do it. I just think it's going to simplify things so much that we don't have to maintain more avatar styles - basically we're choosing to reuse exactly what we do for the avatars in the LHN for #focus mode, and those styles already exist.

Well, the current version matched the group avatar style of non-focus mode, right?

**Comment by shawnborton** [comment]
Yes, but all the other avatars are using the style from #focus mode no? I'm saying, let's just pick the #focus mode styles from the LHN and stick with them in all cases. It should greatly simplify things in that we're just reusing an existing style.

**Comment by dannymcclain** [comment]
>let's just pick the #focus mode styles from the LHN and stick with them in all cases

**Comment by Gonals** [comment]
How does this look?

Screenshot 2024-03-25 at 4 25 05 PM
Screenshot 2024-03-25 at 4 25 20 PM

**Comment by shawnborton** [comment]
Looks good to me! Can you confirm why we have different avatars for the workspace requests though?

**Comment by shawnborton** [comment]
Also, why isn't the truncation here using an ellipsis? Or is the username just "a"?
CleanShot 2024-03-25 at 11 28 30@2x

**Comment by Gonals** [comment]
> Also, why isn't the truncation here using an ellipsis? Or is the username just "a"? CleanShot 2024-03-25 at 11 28 30@2x

The username is `a` 🤣

**Comment by Gonals** [comment]
> Looks good to me! Can you confirm why we have different avatars for the workspace requests though?

Yep:

Screenshot 2024-03-25 at 4 37 52 PM

**Comment by shawnborton** [comment]
Right, I'm saying, why is that avatar where it's just the workspace icon different from how it is displayed in the "Send to" field as well as the LHN item for it?

**Comment by Gonals** [comment]
> Right, I'm saying, why is that avatar where it's just the workspace icon different from how it is displayed in the "Send to" field as well as the LHN item for it?

I'm not quite sure I'm following. It looks like the mockups in the [design](https://docs.google.com/document/d/1Fa-sKBcROi1fjpZnHMUZ3J6Q96yXCEfU-ex14e_wdns/edit#bookmark=id.ipswbj4hafy8) (I think), and, as far as I can tell, it looks like the workspace avatar in the LHN (minus the user's own avatar as a subscript)

**Comment by shawnborton** [comment]
> (minus the user's own avatar as a subscript)

Yes, this is the exact point I am trying to make. Why are we choosing a different avatar for the quick action button? I would think that we should use the identical icon that we use for sending something to a workspace chat, which happens to include the subscript avatar. @trjExpensify do you agree?

**Comment by Gonals** [comment]
> > (minus the user's own avatar as a subscript)
> 
> Yes, this is the exact point I am trying to make. Why are we choosing a different avatar for the quick action button? I would think that we should use the identical icon that we use for sending something to a workspace chat, which happens to include the subscript avatar. @trjExpensify do you agree?

I'm fine with that 🤷 
Screenshot 2024-03-25 at 5 32 50 PM

**Comment by shawnborton** [comment]
Cool, I think it makes sense to be consistent here - basically if we are choosing to show that kind of avatar when you send an expense, we should reuse the same exact one from the shortcut menu too. So that looks good to me given we use that icon style everywhere else.

**Comment by Gonals** [comment]
Neat. All yours @rlinoz!

**Review by rlinoz — APPROVED** [review]

**Comment by Gonals** [comment]
Are we good to merge @shawnborton?

**Comment by shawnborton** [comment]
I think so, yup

File: src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js
Comment (jjcoffee) [inline src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js:155]: We were missing `policy` in the dependencies here, which caused https://github.com/Expensify/App/issues/39045 where the avatar was not kept in sync if the workspace's avatar changed.

**Pull request merged**
